### DISCUSSION TITLE
TeXstudio and highlighting of units syntax (ex, em, mm, cm, pt, ...)

### DISCUSSION THREAD

**Discussion by JosephHallenbeck** [texstudio-org/texstudio#3587] [Q&A] [answered]
I am moving from TeXnicCenter to the TeXstudio and wondering, if there is a way, how to make a custom syntax color for units like (pt, in, mm, cm, ex, em) in TexStudio. TexnicCenter has option for custom syntax of units but TexStudio not. I am trying to customize tex.qnfa without desirable effect. Can anyone help me how to solve this please?

**Comment by dbitouze** [comment]
I don't answer the question but, for the units, I strongly advise the `siunitx` package:
```latex
\documentclass{article}
\usepackage{lmodern}
\usepackage{siunitx}
\begin{document}
When I'm sleeping, I'm running \qty{0}{\km} each \unit{\us}.
\end{document}
```
![image](https://github.com/texstudio-org/texstudio/assets/1032633/6d6540d1-441b-4a15-98ea-905c9d85b815)

TXS provides completion for it.

**Reply by JosephHallenbeck** [reply]
I know about this but I mean slightly different units. I mean highlight units in this case for example 

`\includegraphics[height=6.0em]{........................}`

**Reply by dbitouze** [reply]
Ah, I see. Sorry for the noise.

**Comment by JosephHallenbeck** [comment, marked as answer]
I found the solution which is ok to me. It is adding the custom syntax color for length units (pt, em, mm, cm, pt,...) but also for number in front of the unit. This code needs to be added to the tex.qnfa file:

```
	<list id="keywords/single" format="verbatim">
		<word>[-+.0-9@]+pt+</word>
		<word>[-+.0-9@]+mm+</word>
		<word>[-+.0-9@]+cm+</word>
		<word>[-+.0-9@]+in+</word>
		<word>[-+.0-9@]+ex+</word>
		<word>[-+.0-9@]+em+</word>
		<word>[-+.0-9@]+mu+</word>
		<word>[-+.0-9@]+sp+</word>
	</list>
```
the color of the syntax is linked to the verbatim environment color.
